cli/config.go

```diff
@@ -4,65 +4,286 @@
+func NewConfigCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "config <key> <value>",
+		Short: "CLI local config",
+		Long:  "Local param storage to prevent repetitive passing of keys",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 2 {
+				logUsage(cmd.Use)
+				return
+			}
+
+			key := args[0]
+			value := args[1]
+
+			if err := setConfigValue(key, value); err != nil {
+				logError(err)
+				return
+			}
+
+			logOK()
+		},
 	}
+}
+
+func setConfigValue(key string, value string) error {
+	config, err := read(ConfigPath)
+	if err != nil {
+		return errors.Wrap(errUseExistConf, err)
+	}
+
+	if isURLKey(key) {
+		if ok, err := isValidURL(value); !ok {
+			return errors.Wrap(errInvalidURL, err)
+		}
+	}
+
+	var configKeyToField = map[string]interface{}{
+		"things_url":       &config.Remotes.ThingsURL,
+		"users_url":        &config.Remotes.UsersURL,
+		"reader_url":       &config.Remotes.ReaderURL,
+		"http_adapter_url": &config.Remotes.HTTPAdapterURL,
+		"bootstrap_url":    &config.Remotes.BootstrapURL,
+		"certs_url":        &config.Remotes.CertsURL,
+		"offset":           &config.Filter.Offset,
+		"limit":            &config.Filter.Limit,
+		"name":             &config.Filter.Name,
+		"raw_output":       &config.Filter.RawOutput,
+		"status":           &config.Channel.Status,
+		"state":            &config.Channel.State,
+		"topic":            &config.Channel.Topic,
+		"metadata":         &config.Filter.Metadata,
+		"tls_verification": &config.Remotes.TLSVerification,
+		"msg_content_type": &config.Remotes.MsgContentType,
+	}
+
+	fieldPtr, found := configKeyToField[key]
+	if !found {
+		return errNoKey
+	}
+
+	fieldValue := reflect.ValueOf(fieldPtr).Elem()
+
+	switch fieldValue.Kind() {
+	case reflect.String:
+		fieldValue.SetString(value)
+	case reflect.Int:
+		intValue, err := strconv.Atoi(value)
+		if err != nil {
+			return errors.Wrap(errInvalidInt, err)
+		}
+		fieldValue.SetInt(int64(intValue))
+	case reflect.Bool:
+		boolValue, err := strconv.ParseBool(value)
+		if err != nil {
+			return errors.Wrap(errInvalidBool, err)
+		}
+		fieldValue.SetBool(boolValue)
+	default:
+		return errors.Wrap(errUnsupportedKeyValue, err)
+	}
+
+	buf, err := toml.Marshal(config)
+	if err != nil {
+		return errors.Wrap(errMarshal, err)
+	}
+
+	err = os.WriteFile(ConfigPath, buf, 0644)
+	if err != nil {
+		return errors.Wrap(errWritingConfig, err)
+	}
+
 	return nil
 }
+
+func isValidURL(inputURL string) (bool, error) {
+	u, err := url.Parse(inputURL)
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return false, err
+	}
+
+	return strings.HasPrefix(u.Scheme, "http") || strings.HasPrefix(u.Scheme, "https"), nil
+}
+
+func isURLKey(key string) bool {
+	urlKeys := []string{
+		"things_url",
+		"users_url",
+		"reader_url",
+		"http_adapter_url",
+		"bootstrap_url",
+		"certs_url",
+	}
+
+	for _, urlKey := range urlKeys {
+		if key == urlKey {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
can be replaced by strings.Contains 'url'

Reply:
Will work on that